tangelo/toolboxes/ansatz_generator/tests/test_adapt_ansatz.py

```diff
@@ -57,6 +58,45 @@ def test_adaptansatz_set_var_params(self):
         with self.assertRaises(ValueError):
             ansatz.set_var_params([1.999, 2.999])
 
+    def test_adaptansatz_reference_state_circuit(self):
+        """ Verify variational gate parameters are frozen in a reference state circuit"""
+
+        # Create simple reference circuit:
+        ref_circuit = Circuit([
+            Gate('RY', i, parameter=1.0, is_variational=True)
+            for i in range(4)
+        ])
+
+        # Create ADAPTAnsatz ansatz with reference circuit
+        adapt_ansatz = ADAPTAnsatz(n_spinorbitals=2, n_electrons=2, spin=0,
+                            ansatz_options=dict(reference_state=ref_circuit))
+
+        adapt_ansatz.build_circuit()
+
+        adapt_circ = adapt_ansatz.circuit
+        adapt_circ_gates = list(adapt_circ)
+
+        # Ensure gates are correctly prepended to circuit
+        self.assertEqual(adapt_circ.width,4)
+        self.assertEqual(adapt_circ_gates[0].name, 'RY')
+
+        # Ensure reference circuit gates were correctly converted to
+        # non-variational gates
+        self.assertFalse(adapt_circ_gates[0].is_variational)
+
+        # add qu_op to ansatz
+        adapt_ansatz.add_operator(qu_op)
+
+        # Check ansatz parameters
+        self.assertEqual(adapt_ansatz.n_var_params, 1)
+        self.assertEqual(adapt_ansatz._n_terms_operators, [8])
+
+        # Check that circuit width and first gate is unchanged
+        adapt_circ = adapt_ansatz.circuit
+        adapt_circ_gates = list(adapt_circ)
+        self.assertEqual(adapt_circ.width,4)
```

Comment:
missing space before 4

Reply:
Fixed in [commit 37a9497](https://github.com/goodchemistryco/Tangelo/pull/398/commits/37a94975fdc86946aeb7c4a7e5a0f9a7555aa9b7).